src/com/lftechnology/java/training/niraj/ems/controllers/LoginController.java

```diff
@@ -0,0 +1,76 @@
+package com.lftechnology.java.training.niraj.ems.controllers;
+
+import java.io.Console;
+import java.sql.SQLException;
+import java.util.InputMismatchException;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.lftechnology.java.training.niraj.ems.domains.Employee;
+import com.lftechnology.java.training.niraj.ems.exceptions.CustomException;
+import com.lftechnology.java.training.niraj.ems.services.implementations.RouteServices;
+import com.lftechnology.java.training.niraj.ems.services.implementations.UserServiceImpl;
+
+public class LoginController {
+    private static final Logger LOGGER = Logger.getLogger(LoginController.class.getName());
+
+    private UserServiceImpl userService;
+
+    public LoginController() {
+        userService = new UserServiceImpl();
+    }
+
+    public void login() {
+        int attempt = 0;
+        boolean isAuthenticated = false;
+        Map<String, String> credentials;
+        Employee employee = null;
+        try (Scanner scanner = new Scanner(System.in)) {
+            Console console = System.console();
+            do {
+                if (attempt != 0) {
+                    int attemptLeft = 3 - attempt;
+                    LOGGER.log(Level.WARNING, "\n\n\nYou have {0} attempts left\n\n", attemptLeft);
+                }
+                credentials = userService.getUserCredentials(scanner, console);
+                employee = userService.login(credentials.get("username"), credentials.get("password"));
+                isAuthenticated = employee.getId() != null ? true : false;
+                attempt++;
+            } while (attempt < 3 && !isAuthenticated);
+            if (isAuthenticated) {
+                RouteServices.routeAfterLogin(employee);
+            } else {
+                LOGGER.severe("Incorrect username or password. Exiting the program");
+            }
+        } catch (InputMismatchException | SQLException se) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)